docs/proposals/testing.md

```diff
@@ -0,0 +1,134 @@
+## Testing APB Image
+
+### Motivation
+As the ecosystem of APBs grows we want to facilitate a means for performing a basic sanity check to ensure that an APB is working as the author intended. The basic concept is to package an integration test with the APB code which will contain all of the needed parameters for the actions that the test playbook will run. 
+
+### Limitation Of Proposal
+This proposal and subsequent examples are focusing on testing the provision action. We will be adding other actions in the future. 
+
+The intention of this design is to check that an APB passes a basic sanity check before publishing to the service catalog. The initial proposal is to be used by CI or another process to check the APB before it is published. This proposal is not meant to be testing a live service. OpenShift provides the ability to test a live service using [liveness and readiness probes](https://docs.openshift.org/latest/dev_guide/application_health.html), which you can add when provisioning. 
+
+
+### Design
+The base APB entry point will be able to find and run the test action. The test action will be a user defined playbook. 
+
+* To include the testing of an APB just add the playbook `test.yml`
```

Comment:
I see value in splitting out the requirement we are imposing as well as the suggestions for best practices.

Requirement:
 - test.yml exists and is intended to run provision logic at a minimal with some known values for a basic configuration.  Requirement is that "oc run imagename test" is sufficient to execute the APB on the users cluster (perhaps we need to specify a namespace and syntax a little so it is accurate)

If test.yml succeeds, assume test was successful.
If test.yml fails, assume failure of test.


APB author is free to structure the test logic as they deem fit, we will provide suggestions of what we recommend yet they have freedom in adapting as long as they confirm to the above.



Recommendations/Best Practices:
 - In addition to the minimal requirements of a test.yml, we recommend a verify role be used to contain logic for checking if the service provisioned is working as expected.
 - Then continue with any other recommendations

I would personally not cover much in best practices/suggestions at this phase as I assume we will learn what patterns work best in the test as we flesh out a few examples.  I think it's good info to capture, just wonder if it might be premature for now.